ts/features/pn/components/MessageDetails.tsx

```diff
@@ -1,193 +1,60 @@
-      <ScrollView
-        contentContainerStyle={{
-          flexGrow: 1,
-          padding: IOVisualCostants.appMarginDefault
-        }}
-        ref={scrollViewRef}
+    <ScrollView>
+      <MessageDetailHeader
+        service={service}
+        sender={message.senderDenomination}
+        subject={message.subject}
+        createdAt={message.created_at}
       >
-        {service && <MessageDetailsHeader service={service} />}
-        <VSpacer />
-        <MessageDetailsContent message={message} />
-        {isCancelled && (
-          <>
-            <VSpacer />
-            <DSFullWidthComponent>
-              <StatusContent
-                accessibilityLabel={I18n.t(
-                  "features.pn.details.cancelledMessage.body"
-                )}
-                backgroundColor={statusColorMap.warning}
-                foregroundColor={getStatusTextColor(LevelEnum.warning)}
-                iconName={statusIconMap.warning}
-                testID={"PnCancelledMessageBanner"}
-                ref={viewRef}
-              >
-                {I18n.t("features.pn.details.cancelledMessage.body")}
-              </StatusContent>
-            </DSFullWidthComponent>
-          </>
-        )}
-
-        {RA.isNonEmpty(attachmentList) && (
-          <MessageDetailsSection
-            title={I18n.t("features.pn.details.attachmentsSection.title")}
-          >
-            <LegacyMessageAttachments
-              disabled={isCancelled}
-              attachments={attachmentList}
-              downloadAttachmentBeforePreview={true}
-              openPreview={openAttachment}
-            />
-          </MessageDetailsSection>
-        )}
-        <MessagePayments
-          messageId={messageId}
-          isCancelled={isCancelled}
-          payments={payments}
-          completedPaymentNoticeCodes={completedPaymentNoticeCodes}
-          maxVisiblePaymentCount={maxVisiblePaymentCount}
-          presentPaymentsBottomSheetRef={presentPaymentsBottomSheetRef}
-        />
-
-        {!isCancelled && RA.isNonEmpty(f24List) ? (
-          <>
-            <MessageF24 attachments={f24List} openPreview={openAttachment} />
-            <VSpacer size={24} />
-          </>
-        ) : null}
-
-        <MessageDetailsSection
-          title={I18n.t("features.pn.details.infoSection.title")}
-        >
-          <ListItemInfoCopy
-            value={message.iun}
-            onPress={() => clipboardSetStringWithFeedback(message.iun)}
-            accessibilityLabel={I18n.t("features.pn.details.infoSection.iun")}
-            label={I18n.t("features.pn.details.infoSection.iun")}
-          />
-          <H5 color="bluegrey">
-            {I18n.t("features.pn.details.timeline.title")}
-          </H5>
-          <VSpacer size={24} />
-          <MessageTimeline
-            message={message}
-            onExpand={() => {
-              scrollViewRef.current?.scrollToEnd({ animated: true });
-            }}
+        <View style={IOStyles.row}>
```

Comment:
Instead of using children, should we move this inside the `MessageDetailHeader`

Reply:
since the number of tags can vary between SEND messages and generic ones, it could be useful to pass the tags as children prop. what do you think?